### DISCUSSION TITLE
How can I make the total row count unaffected on rowDetailView click?

### DISCUSSION THREAD

**Discussion by LRHF** [ghiscoding/aurelia-slickgrid#1084] [Q&A]
Hi again,
I have a paginated grid with 10 items in total.
![image](https://github.com/ghiscoding/aurelia-slickgrid/assets/22406500/42c4fa30-f554-4daa-ad0a-6f8a32d72028)

I added a `rowDetailView` with `panelRows = 5` but when expanding the row details, the total row count shows 14 which is very strange and sounds like a bug.

![image](https://github.com/ghiscoding/aurelia-slickgrid/assets/22406500/c22c6acd-4a0c-4d82-a20d-0b41cec8cd42)

How can I make that value unaffected?

I tired with this fix which is working right now, but I don't know if that is the right thing to do. 
```
onCellClicked(event, args) {
  this.aureliaGrid.paginationService!.dispose();
};
```  
I added pagination to the example 19 and shows a similar issue, total items are 1000, but after clicking on details it shows 1007.

Thanks!

**Comment by ghiscoding** [comment]
Row Detail was not created to work with Pagination, it is not at all supported and you will eventually have problem with it, for example this SlickGrid [issue](https://github.com/6pac/SlickGrid/issues/718).

**Comment by LRHF** [comment]
I see. Well, I need to implement a grid with tree view and pagination, like [this one](https://ej2.syncfusion.com/angular/demos/?_gl=1*1rv8a10*_ga*MjAwODM2MzI0NS4xNjk1MDczODY0*_ga_R3Q3BGFFC9*MTY5NjA0MjU5MS41LjEuMTY5NjA0MjkwNi40Mi4wLjA.*_ga_41J4HFMX1J*MTY5NjA0MjczNS40LjEuMTY5NjA0MjkwNy4wLjAuMA..&_ga=2.251928585.1881771228.1695978394-2008363245.1695073864#/material/treegrid/default) (Syncfusion for Angular). 

So I started with the SlickGrid Tree Data, created all the structure, and it was working great. But when I added pagination, I saw the message in console saying they are not compatible. That was a shame.

So now I moved the rowDetailView with pagination and it's working fine (I know counter and total are wrong, but I can hide them).
I'm using a trick to increase the size of the grid which solve some of the issues you mentioned in the link, since the DetailView panel is always visible now:
```
onCellClicked(event, args) {
    const currentItemPerPage = this.aureliaGrid.paginationService.getCurrentItemPerPage();
    this.aureliaGrid.paginationService!.changeItemPerPage(currentItemPerPage + PANEL_ROWS - 1);
}; 
```
But I need to do something similar on collapsing. Is there any events or callbacks that I can use on Detail View expand and collapse?
Also, I will try to set `panelRows` dynamically, so that the children can fit into the DetailView panel.

**Reply by ghiscoding** [reply]
You're trying to mix too many of these features together, Pagination is just a pain to work with and is not working (neither supported) with Row Detail or Tree Data. It's already hard to get it working correctly without Pagination, but if you then add Pagination it just become even more complex. You're comparing a paid and expensive product against a free product, so of course Syncfusion and Ag-Grid are better product but they cost a lot too. Let's put it this way, Pagination is problematic with many other features, it doesn't work well with Row Detail, Tree Data, Grouping and probably more, it is simply best to use without any other features.

The events that are available should be showing with intellisense, but anyway here's the [rowDetailView.interface](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/interfaces/rowDetailView.interface.ts) and for the Tree Data, the events are shown in Example 27. I didn't have time to add them to the Wikis it seems

https://github.com/ghiscoding/aurelia-slickgrid/blob/bce0099140ba5d87a3e7f334644607a0231b95d9/src/examples/slickgrid/example27.html#L89-L94
